presto-orc/src/test/java/com/facebook/presto/orc/TestSelectiveOrcReader.java

```diff
@@ -288,6 +289,81 @@ public void testFloats()
                                 1, FloatRange.of(1.0f, false, true, 7.0f, false, true, true))));
     }
 
+    @Test
+    public void testOrdering()
+            throws Exception
+    {
+        // We can now check that filters are called in the order we expect.
+        // Watch out for:
+        // - If you efficiently filter out all positions in a page by filter x, then filter x + 1 won't even be called, making your test fail
+        // - Only works for bigintrange and doublerange right now
+
+        // Setup
+
+        Random random = new Random(0);
+
+        List<Integer> values1 = newArrayList(limit(cycle(ImmutableList.of(1, 3, 5, 7, 11)), 200));
+        List<Integer> values2 = IntStream.range(0, 200).boxed().collect(toList());
+        Collections.shuffle(values2, random);
+        List<Double> values3 = doubleSequence(0, 0.1, 200);
+        List<List<Integer>> values4 = IntStream.range(0, 200)
+                .map(i -> 1 + random.nextInt(10))
+                .mapToObj(size -> makeArray(size, random, Optional.of(5)))
+                .collect(toImmutableList());
+
+        BigintRange firstFilter = BigintRange.of(1, 1, true);
+        BigintRange secondFilter = BigintRange.of(2, 200, true);
+        DoubleRange thirdFilter = DoubleRange.of(3, false, false, 200, false, false, true);
+        BigintRange fourthFilter = BigintRange.of(4, 4, false);
+
+        // # 1. Test order of columns
+
+        tester.testRoundTripTypesWithOrder(ImmutableList.of(INTEGER, INTEGER),
+                ImmutableList.of(values1, values2),
+                toSubfieldFilters(
+                        ImmutableMap.of(
+                                0, firstFilter,
+                                1, secondFilter)),
+                ImmutableList.of(0, 1));
+
+        // # 2. Test the reverse order of filter functions inputted -- we should still get the same order of column reads
```

Comment:
I'd change the framework so that when filter order is specified it runs the test twice: first using the specified column order, second using reversed column order. This would simplify the test code and ensure that order is aways tested properly.

Reply:
okay!